kybra/compiler/kybra_post_install/build.rs

```diff
@@ -0,0 +1,154 @@
+
+fn main() -> Result<(), String> {
+    let ic_build = get_and_set_ic_build();
+
+    if ic_build == false {
+        return Ok(());
+    }
+
+    let kybra_version = get_kybra_version()?;
+    let home_dir = dirs::home_dir().ok_or(create_error_string("Home directory not found"))?;
+
+    let python_stdlib_binary_path =
+        home_dir.join(format!(".config/kybra/{kybra_version}/bin/python_stdlib"));
+    handle_python_stdlib_exists(&python_stdlib_binary_path);
+
+    let python_stdlib_src_path = home_dir.join(format!(".config/kybra/{kybra_version}/Lib"));
+    let kybra_post_install_src_lib_path = Path::new("src/Lib");
+
+    if kybra_post_install_src_lib_path.exists() {
+        clean_kybra_post_install_src_lib(&kybra_post_install_src_lib_path)?;
+        return Ok(());
+    }
+
+    copy_from_python_stdlib_src_to_kybra_post_install_src(
+        &python_stdlib_src_path,
+        &kybra_post_install_src_lib_path,
+    )?;
+
+    let github_repo = "https://github.com/RustPython/RustPython.git";
+    let commit_hash = "f12875027ce425297c07cbccb9be77514ed46157";
+
+    let temp_dir = tempdir().map_err(|e| error_to_string(&e))?;
+    git_clone(github_repo, &temp_dir)?;
+    git_checkout(commit_hash, &temp_dir)?;
+
+    let git_repo_lib_dir = temp_dir.path().join("Lib");
+    copy_directory(&git_repo_lib_dir, &python_stdlib_src_path)?;
+    copy_directory(&python_stdlib_src_path, &kybra_post_install_src_lib_path)?;
+
+    temp_dir.close().map_err(|e| error_to_string(&e))?;
+
+    Ok(())
+}
+
+fn get_kybra_version() -> Result<String, String> {
+    std::env::var("KYBRA_VERSION")
+        .map_err(|_| create_error_string("KYBRA_VERSION environment variable is not present"))
+}
+
+fn handle_python_stdlib_exists(python_stdlib_binary_path: &Path) {
+    if python_stdlib_binary_path.exists() {
+        println!("cargo:rustc-cfg=python_stdlib_exists");
+    }
+}
+
+fn clean_kybra_post_install_src_lib(kybra_post_install_src_lib_path: &Path) -> Result<(), String> {
+    fs_extra::dir::remove(kybra_post_install_src_lib_path).map_err(|e| error_to_string(&e))?;
+    Ok(())
+}
+
+fn copy_from_python_stdlib_src_to_kybra_post_install_src(
+    python_stdlib_src_path: &Path,
+    kybra_post_install_src_lib_path: &Path,
+) -> Result<(), String> {
+    if python_stdlib_src_path.exists() {
+        copy(
+            python_stdlib_src_path,
+            kybra_post_install_src_lib_path,
+            &CopyOptions {
+                copy_inside: true,
+                ..Default::default()
+            },
+        )
+        .map_err(|e| error_to_string(&e))?;
+    }
+    Ok(())
+}
+
+fn git_clone(github_repo: &str, temp_dir: &tempfile::TempDir) -> Result<(), String> {
+    let status = Command::new("git")
+        .args(&[
+            "clone",
+            github_repo,
+            temp_dir
+                .path()
+                .to_str()
+                .ok_or(create_error_string("Could not convert path to &str"))?,
```

Comment:
I would say something more like, Unable to clone path `{}`: could not convert to &str.